WordPress/wporg-developer
# Update Since page layout

Closes #263 

Depends on https://github.com/WordPress/wporg-parent-2021/pull/112 for button and checkbox toggle button styles.

Updates the taxonomy page for `wp-parser-since` to match the [design](https://www.figma.com/file/2WxlJFzMJvqPfZL1EkAOVp/Developer-Resources?node-id=2934%3A6854&mode=dev).

### Changes

- Add Search field
- Split existing filters into seperate release selector and post type filters
- Change the query title prefix to 'New and updated in ...'
- Minor font style and spacing updates in the post content items

The only complex part of this was reworking the post type filters; adding the 'All' control and restyling the checkboxes to display like buttons. I initially tried to add the 'All' control as another checkbox with no value, but submitting the form resulted in a 404 as the post_type query var was empty. My solution is to use a button which simply clears the other checkboxes so that the form is submitted with no post_type query vars. The logic for all this is in a new [view script](https://github.com/WordPress/wporg-developer/pull/292/files#diff-df917505c8b2f284644e1b9b2c5beb3e5c04c061664bb79d80710a7be6ca0f3a).

## Screenshots

| Desktop | Tablet | Mobile |
|-|-|-|
| ![localhost_8888_reference_since_6 1 0_page_2__wp-parser-since=6-1-0(Desktop)](https://github.com/WordPress/wporg-developer/assets/1017872/2e38dc3a-378e-4bec-9d7b-8fb8b0177961) | ![localhost_8888_reference_since_6 1 0_page_2__wp-parser-since=6-1-0(iPad)](https://github.com/WordPress/wporg-developer/assets/1017872/ef63364f-3e7b-4bd7-a51a-2f08e7ddee94) | ![localhost_8888_reference_since_6 1 0_page_2__wp-parser-since=6-1-0(Samsung Galaxy S20 Ultra)](https://github.com/WordPress/wporg-developer/assets/1017872/cbfe487c-0678-44da-a8dd-5f5c24418130) |

| State | |
|-|-|
| Filters checked | ![Screenshot 2023-09-28 at 2 06 00 PM](https://github.com/WordPress/wporg-developer/assets/1017872/d5374825-5953-4f91-a2ce-6f1ab257cdb1) |
| Hover filters and buttons | ![Screenshot 2023-09-28 at 1 57 33 PM](https://github.com/WordPress/wporg-developer/assets/1017872/87904a88-eedb-4eee-bf1f-3e83a6672e76) |
| Focus filters and buttons | ![Screenshot 2023-09-28 at 1 58 22 PM](https://github.com/WordPress/wporg-developer/assets/1017872/d64e2e02-1243-49e3-9f2b-20de032e3020) |
| Active filters and buttons | ![Screenshot 2023-09-28 at 2 05 43 PM](https://github.com/WordPress/wporg-developer/assets/1017872/ca06490b-fdc3-4601-b0d9-7047ee8e980f) |

## Testing

1. Use wp-env to set your parent theme to the [button styles branch](https://github.com/WordPress/wporg-parent-2021/pull/112)
2. Navigate to http://localhost:8888/reference/since/6.3.0/
3. Test changing release version using the dropdown
4. Test filtering the results by post type, including using the 'All' button
5. Test with keyboard and screen reader
6. Check interactive states, which are based on [Button/Small in the Design Library](https://www.figma.com/file/HOJTpCFfa3tR0EccUlu0CM/WordPress.org-Design-Library?type=design&node-id=2710-958&mode=design&t=pXEyBCRAQ1yeAbV4-0)
